src/hotspot/cpu/ppc/sharedRuntime_ppc.cpp

```diff
@@ -3246,8 +3246,8 @@ void SharedRuntime::montgomery_square(jint *a_ints, jint *n_ints,
   // Make very sure we don't use so much space that the stack might
   // overflow. 512 jints corresponds to an 16384-bit integer and
   // will use here a total of 6k bytes of stack space.
+  guarantee(longwords <= 341, "must be");
```

Comment:
I was mystified about this number.  Can you make this instead like:

int thing = sizeof (unsigned long) * 3;
guarantee(longwords <= 8192 / thing, "must be");

Or is division just as unsafe?

Should this be uint64_t, not long ?  Or change it to julong if it's a small change to match the x86 version.

Reply:
How about just:    guarantee(longwords <= 8092 / (sizeof(julong) * 3), "must be");   ?
I don't think that doing division is a problem.